src/lsm/table_memory.zig

```diff
@@ -40,6 +40,22 @@ pub fn TableMemoryType(comptime Table: type) type {
             },
         };
 
+        // `sort_suffix` breaks the `values` array into “sorted runs” (sorted sub‑arrays).
+        // Each SortedRun captures the start (min) and end (max) indices of one sorted run.
+        // We can exploit this knowledge to check if the full array is sorted in O(k) and avoid
+        // the final sort, which reduces tail latency.
+        // The number of calls to `sort_suffix` are determined by `constants.lsm_compaction_ops`.
+        const SortedRun = struct {
+            min: u32,
+            max: u32,
```

Comment:
```suggestion
            min: u32, // inclusive
            max: u32, // exclusive
```

We sometimes use both-including, like HeaderRange in journal or view Range in replica